Algorithms/DataCompression/huffman.cs

```diff
@@ -27,184 +28,298 @@ public Huff(Huff leftChild, Huff rightChild)
+            var man = new Man();
+
+            // generates and displays the huffman code
+            Console.WriteLine("\nHuffman Code:");
+            GenerateCode(parentNode1, "", man);
+
+            return man;
+        }
+
+        /// <summary>
+        /// Find the frequency for each caracter on the text.
+        /// </summary>
+        /// <returns>Updateds frequency Array</returns>
+        private void GetCharFrequencies()
+        {
+            var count = 1;
+
+            for (var i = 0; i < Len; i++)
+            {
+                for (var j = i + 1; j < Len; j++)
                 {
-                    if (c[i] == c[j])
+                    if (_c[i] == _c[j])
                     {
                         count++;
                     }
                 }
-                d[i] = count;
+
+                _d[i] = count;
                 count = 1;
             }
-            for (var i = 0; i < n; i++)
+        }
+
+        private void FlagCharsByFrequency()
+        {
+            var flag = false;
+
+            for (var i = 0; i < Len; i++)
             {
-                for (var j = 0; j < n; j++)
+                for (var j = 0; j < Len; j++)
                 {
-                    if (c[i] == c1[j])
+                    if (_c[i] == _c1[j])
                     {
                         flag = true;
                     }
                 }
                 if (!flag)
                 {
-                    c1[pos] = c[i];
-                    d1[pos] = d[i];
-                    pos++;
+                    _c1[Pos] = _c[i];
+                    _d1[Pos] = _d[i];
+                    Pos++;
                 }
                 flag = false;
             }
-            for (var i = 0; i < pos; i++)
+        }
+
+        /// <summary>
+        /// Prints to console the input array
+        /// </summary>
+        /// <param name="arr">Input array</param>
+        /// <param name="inverse">To print starting at last index/</param>
+        private void PrintArray(Array arr, bool inverse = false)
+        {
+            if (inverse)
             {
-                Console.Write("{0}\t", c1[i]);
+                for (var i = Pos - 1; i >= 0; i--)
+                {
+                    Console.Write("{0}\t", arr.GetValue(i));
+                }
             }
-            Console.WriteLine();
-            for (var i = 0; i < pos; i++)
+            else
             {
-                Console.Write("{0}\t", d1[i]);
+                for (var i = 0; i < Pos; i++)
+                {
+                    Console.Write("{0}\t", arr.GetValue(i));
+                }
             }
-            int temp;
-            char ch;
 
-            // sorts array d1 (frequencies) and array c1
-            for (var i = 0; i < pos; i++)
+            Console.WriteLine();
+        }
+
+        /// <summary>
+        /// Sorts array frequencies(d1) array and char array (c1).
+        /// </summary>
+        private void SortArrays()
```

Comment:
If i got it right, this method sorts arrays. I suppose we can get ISorter<int> interface as a constructor parameter and use it to sort arrays.

Reply:
Arrays get sorted in a specific way, I'll try to see if there's a way to use any of the current sorters.

Reply:
Okay, I will do it later.